src/Components/Web.JS/src/DomWrapper.ts

```diff
@@ -22,6 +23,12 @@ function focus(element: HTMLOrSVGElement, preventScroll: boolean): void {
   }
 }
 
+function focusOnNavigate(selector: string): void {
+  if ( !(location.hash.length > 1 && elementExists(location.hash.slice(1))) ) {
+    focusBySelector(selector);
+  }
```

Comment:
It looks like this logic suppresses `FocusOnNavigate` if the URL contains a hash. I haven't actually tried this out so am not sure what the exact actual behavior is but in general would expect `focus` to be independent to `scrollIntoView`.

Does it work to leave `FocusOnNavigate` to do its existing behavior regardless of the URL hash?

The reason is that `FocusOnNavigate` is an accessibility feature that causes a page title to be announced. I would expect people still want that to happen even if we're also auto-scrolling to some unrelated element based on the URL hash.

Reply:
> It looks like this logic suppresses FocusOnNavigate if the URL contains a hash.
Yes it suppresses if the url contains hash and the element is in the page. 

> Does it work to leave FocusOnNavigate to do its existing behavior regardless of the URL hash?

No, the focus script is called after the scroll script so when navigating to another url with hash doesn't work.

Reply:
I don't get why `FocusOnNavigate` needs to be affected by this. Why do the following two things interact?

1. The `<FocusOnNavigate>` component performs a JS interop call after certain renders that results in a call to `someDomElement.focus()`
2. The new logic you've added in `Renderer.ts` runs after each renderbatch and may result in a call to `someOtherDomElement.scrollIntoView()`

What causes these to interfere with each other? Focus and scroll are two different things so I would expect them to work independently.

Sorry if I'm missing something obvious!

Reply:
`focus()` also affects the scroll position. https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus
I think I'd better use `preventScroll` option rather than not focusing at all. Thanks!